src/dbnode/storage/index/fields_terms_iterator.go

```diff
@@ -278,14 +282,14 @@ func (fti *fieldsAndTermsIter) Err() error {
 	return fti.err
 }
 
-func (fti *fieldsAndTermsIter) Close() error {
+func (fti *fieldsAndTermsIter) Close(ctx context.Context) error {
 	var multiErr xerrors.MultiError
 	if fti.fieldIter != nil {
 		multiErr = multiErr.Add(fti.fieldIter.Close())
 	}
 	if fti.termIter != nil {
 		multiErr = multiErr.Add(fti.termIter.Close())
 	}
-	multiErr = multiErr.Add(fti.Reset(nil, fieldsAndTermsIteratorOpts{}))
+	multiErr = multiErr.Add(fti.Reset(ctx, nil, fieldsAndTermsIteratorOpts{}))
```

Comment:
it's really weird that `Reset(nil)` doesn't blow up

Reply:
maybe a confusion API but seems like the idea is that to reset it, you nil out the reader

Reply:
but it uses the reader inside Reset to perform a search lol

Reply:
haha yeah...